Stylis v4 (#1817)
emotion-js/emotion: .changeset/warm-ties-drop.md

It has been completely rewritten and comes with some breaking changes. Most notable ones that might affect Emotion users are:

- plugins written for the former Stylis v3 are not compatible with the new version. To learn more on how to write a plugin for Stylis v4 you can check out its [README](https://github.com/thysultan/stylis.js#middleware) and the source code of core plugins.
- vendor-prefixing was previously customizable using `prefix` option. This was always limited to turning off all of some of the prefixes as all available prefixes were on by default. The `prefix` option is gone and to customize which prefixes are applied you need to fork (copy-paste) the prefixer plugin and adjust it to your needs. While this being somewhat more problematic to setup at first we believe that the vast majority of users were not customizing this anyway. By not including the possibility to customize this through an extra option the final solution is more performant because there is no extra overhead of checking if a particular property should be prefixed or not.

> - Prefixer is now just a plugin which happens to be put in default `stylisPlugins`. If you plan to use custom `stylisPlugins` and you want to have your styles prefixed automatically you must include prefixer in your custom `stylisPlugins`.

Reviewer: You should tell users where to get the default prefixer from here.

Author: Added such mention just now.

> - `@import` rules are no longer special-cased. The responsibility to put them first has been moved to styles author. They also can't be nested within other rules now - this excludes them being usable within `css`. It's only possible to write them at the top level of global styles.

Reviewer: ```suggestion
- `@import` rules are no longer special-cased. The responsibility to put them first has been moved to the author of the styles. They also can't be nested within other rules now - this means they are not usable within `css`. It's only possible to write them at the top level of global styles.
```

> this means they are not usable within `css`.

This is wrong, no?

```jsx
<Global styles={css`@import whatever it is`} />
```

Author: Right, was more thinking about vanilla Emotion where this would not be possible, but you are right - `css` from `@emorion/react` can be fed to `<Global/>` so this wasn't true. I've removed this part of a changeset.
